jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/error/ErrorManager.java

```diff
@@ -0,0 +1,149 @@
+package com.tmobile.cso.pacman.datashipper.error;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tmobile.cso.pacman.datashipper.config.CredentialProvider;
+import com.tmobile.cso.pacman.datashipper.util.Constants;
+
+/**
+ * The Class AWSErrorManager.
+ */
+public abstract class ErrorManager implements Constants {
+	
+	/** The Constant LOGGER. */
+	private static final Logger LOGGER = LoggerFactory.getLogger(ErrorManager.class);
+	
+	/** The s 3 account. */
+	private String s3Account = System.getProperty("base.account");
+	
+	/** The s 3 region. */
+	private String s3Region = System.getProperty("base.region");
+	
+	/** The s 3 role. */
+	private String s3Role =  System.getProperty("s3.role");
+	
+	/** The bucket name. */
+	private String bucketName =  System.getProperty("s3");
+	
+	/** The data path. */
+	private String dataPath =  System.getProperty("s3.data");
+	
+	/** The error info. */
+	private Map<String,List<Map<String,String>>> errorInfo ;
+	
+	protected String dataSource;
+	
+	/** The error manager. */
+	private static ErrorManager errorManager ;
+	
+	/**
+	 * Instantiates a new AWS error manager.
+	 */
+	
+	
+	/**
+	 * Gets the single instance of AWSErrorManager.
+	 *
+	 * @return single instance of AWSErrorManager
+	 */
+	public static ErrorManager getInstance(String dataSource){
+		if(errorManager==null){
+			switch(dataSource) {
+				case "aws":
+					errorManager = new AwsErrorManager(); 
+					errorManager.dataSource ="aws";
+					break;
+				case "azure":
+					errorManager = new AzureErrorManager(); 
+					errorManager.dataSource ="azure";
+					break;
+				default:
+			}
+			
+		}
+		return errorManager;
+	}
+	
+	/**
+	 * Fetch error info.
+	 *
+	 * @param datasource the datasource
+	 * @param errorList the error list
+	 */
+	private void fetchErrorInfo(List<Map<String,String>> errorList){
+		if(errorInfo==null){
+			ObjectMapper objectMapper = new ObjectMapper();
+	    	List<Map<String, String>> inventoryErrors = new ArrayList<>();
+	    	AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
+	                .withCredentials(new AWSStaticCredentialsProvider(new CredentialProvider().getCredentials(s3Account,s3Role))).withRegion(s3Region).build();
+	    	try {
+		    	S3Object inventoryErrorData = s3Client.getObject(new GetObjectRequest(bucketName,dataPath+"/"+dataSource+"-loaderror.data"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code retrieves and decodes the contents of an S3 object to a String.
This can be done by calling [getObjectAsString](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3.html#getObjectAsString-java.lang.String-java.lang.String-). Using this function simplifies the code and eliminates the need to close the S3 object once it is read.